lusSTR/scripts/repeat.py

```diff
@@ -163,15 +163,16 @@ def repeat_copy_number(bf, repeat):
     The input is a sequence string collapsed to bracketed sequence form.
     """
     longest = 0
-    for block in bf.split(" "):
-        if block == repeat:
-            if 1 > longest:
-                longest = 1
-        match = re.match(r"\[" + repeat + r"\](\d+)", block)
-        if match:
-            length = int(match.group(1))
-            if length > longest:
-                longest = length
+    if bf != "":
+        for block in bf.split(" "):
+            if block == repeat:
+                if 1 > longest:
+                    longest = 1
+            match = re.match(r"\[" + repeat + r"\](\d+)", block)
+            if match:
+                length = int(match.group(1))
+                if length > longest:
+                    longest = length
```

Comment:
Code is dropping these garbage sequences.

Reply:
This was a simple and straightforward change, but I have a suggestion for improvement: instead of indenting the code in a conditional block, short-circuit the inverse condition.

```python
if bf == "":
    return 0
for block in bf.split(" "):
   ### blah blah blah
```

This has a few benefits. First of all, you see what happens right away if `bf` is empty: in the current code, you have to read all the way to the end of the block to find out that...nothing happens. Second, it keeps the nesting and indentation to a minimum: this is not only good for legibility, it also (at least in this case) makes for cleaner diffs in code review. My suggestion above would result in a simple two-line diff: the diff for the current code looks like this.

<img width="1689" alt="Screenshot 2024-01-11 at 9 39 33 PM" src="https://github.com/bioforensics/lusSTR/assets/566823/edcf72ba-d450-4fe7-a99d-c4342afb692e">

Reply:
makes sense! I'll update.